# Master-s-Thesis-Scripts
The scripts used for my Master's thesis are grouped by topic or test.

- *Population_Selection* Contains the script for defining the populations to include in the analyses.
- *Admixture* Contains the script to convert genotypes to Plink format, the parameters to run the program, the script to run ADMIXTURE, and one to visualise cross-validation and admixture results.
- *F_tests* Contains scripts to run F-statistics with qp3Pop (_f3.sh_) and qpDstat(_f4.sh_) and the visualisation of F4 results.
- *Genotuping* Contains the script to run variant calling and to count the SNPs obtained.
- *PCA* Contains the scripts to run worldwide PCA with smartpca and for plotting PCA results
- *Merge* Contains the scripts to merge Umingmak results from the same treatment and to merge the Umingmak genotypes with the 1240K dataset.
- *Schmutzi* Contains the script to run Schmutzi on the mitochondrial DNA (_Schmutzi_Pipeline.sh_).
- *SexDet_XCont* Contains the script to run sex determination and X contamination and the code to calculate chromosome's rate (_sexDetermination.hg19.awk_).

